src/Tasks/CheckExternalLinksTask.php

```diff
@@ -0,0 +1,230 @@
+        }
+
+        // Check if we need to update CSS class, otherwise return
+        if ($markedBroken == $foundBroken) {
+            return;
+        }
+        if ($foundBroken) {
+            $class .= ' ss-broken';
+        } else {
+            $class = preg_replace('/\s*\b(ss-broken)\b\s*/', ' ', $class);
+        }
+        $link->setAttribute('class', trim($class));
+    }
+
+    /**
+     * Determine if the given HTTP code is "broken"
+     *
+     * @param int $httpCode
+     * @return bool True if this is a broken code
+     */
+    protected function isCodeBroken($httpCode)
+    {
+        // Null represents no request attempted
+        if ($httpCode === null) {
+            return false;
+        }
+
+        // do we have any whitelisted codes
+        $ignoreCodes = Config::inst()->get('CheckExternalLinks', 'IgnoreCodes');
+        if (is_array($ignoreCodes) && in_array($httpCode, $ignoreCodes)) {
+            return false;
+        }
+
+        // Check if code is outside valid range
+        return $httpCode < 200 || $httpCode > 302;
+    }
+
+    /**
+     * Runs the links checker and returns the track used
+     *
+     * @param int $limit Limit to number of pages to run, or null to run all
+     * @return BrokenExternalPageTrackStatus
+     */
+    public function runLinksCheck($limit = null)
+    {
+        // Check the current status
+        $status = BrokenExternalPageTrackStatus::get_or_create();
+
+        // Calculate pages to run
+        $pageTracks = $status->getIncompleteTracks();
+        if ($limit) {
+            $pageTracks = $pageTracks->limit($limit);
+        }
+
+        // Check each page
+        foreach ($pageTracks as $pageTrack) {
+            // Flag as complete
+            $pageTrack->Processed = 1;
+            $pageTrack->write();
+
+            // Check value of html area
+            $page = $pageTrack->Page();
+            $this->log("Checking {$page->Title}");
+            $htmlValue = Injector::inst()->create('HTMLValue', $page->Content);
+            if (!$htmlValue->isValid()) {
+                continue;
+            }
+
+            // Check each link
+            $links = $htmlValue->getElementsByTagName('a');
+            foreach ($links as $link) {
+                $this->checkPageLink($pageTrack, $link);
+            }
+
+            // Update content of page based on link fixes / breakages
+            $htmlValue->saveHTML();
+            $page->Content = $htmlValue->getContent();
+            $page->write();
+
+            // Once all links have been created for this page update HasBrokenLinks
+            $count = $pageTrack->BrokenLinks()->count();
+            $this->log("Found {$count} broken links");
+            if ($count) {
+                // Bypass the ORM as syncLinkTracking does not allow you to update HasBrokenLink to true
+                DB::query(sprintf(
+                    'UPDATE "SiteTree" SET "HasBrokenLink" = 1 WHERE "ID" = \'%d\'',
```

Comment:
Ideally `SiteTree` would be retrieved from `DataObject::getSchema()->tableName(SiteTree::class)` in case it changes in future (to ease future upgrade paths)